libcnb/src/runtime.rs

```diff
@@ -138,14 +140,32 @@ pub fn libcnb_runtime_detect<B: Buildpack>(
         buildpack_descriptor: read_buildpack_descriptor()?,
     };
 
-    match buildpack.detect(detect_context)?.0 {
-        InnerDetectResult::Fail => Ok(exit_code::DETECT_DETECTION_FAILED),
+    #[cfg(feature = "trace")]
+    let mut trace = start_trace(&detect_context.buildpack_descriptor.buildpack, "detect");
+
+    let detect_result = buildpack.detect(detect_context);
+
+    #[cfg(feature = "trace")]
+    if let Err(ref err) = detect_result {
```

Comment:
There's a bunch of code that looks like this. I'm really looking forward to `inspect_err` becoming usable: https://github.com/rust-lang/rfcs/issues/3190.

Reply:
Looks very close to being stabilised at least 😄 
https://github.com/rust-lang/rust/pull/116866